### ISSUE TITLE
prefer-grants-property: remove the temporary receiver guards once corsa-oxlint resolves the remaining receiver forms

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#598]
### Summary

PR #587 added two temporary guards to the oxlint `prefer-grants-property` rule (`packages/oxlint/src/core/ts-type/finder/receiver-type.ts`) so that the plugin does not report on a receiver whose type corsa-oxlint mis-resolves:

- a computed `MemberExpression` is treated as unresolvable when the member expression and its object resolve to the same type (the symptom of corsa returning the indexed object's type);
- a call whose callee has two or more call signatures with differing return types is treated as unresolvable (corsa returns the last declared overload's return type when the overloads differ only in literal parameter types).

Both guards exist only because of upstream defects reported in ubugeeei-prod/corsa-bind#490 (index-signature access, non-literal index access, `TSNonNullExpression`, literal-discriminated overloads, generic calls). Once a corsa-oxlint release containing the fix is pinned in `packages/oxlint/package.json`, the guards stop firing and must be removed so that the oxlint rule resolves receivers exactly like the eslint rule.

### Tasks

1. Bump `corsa-oxlint` to the release that resolves the forms listed in corsa-bind#490, and re-run a dependency-free probe (`corsa-oxlint/rule-tester`) to confirm each form now resolves to the expression's own type (literal and non-literal index on arrays / tuples / `Record`, class and interface index signatures, `make()!`, `targets[0]!`, literal-discriminated overloads, `id<T>(x)`).
2. Remove `findReceiverType`'s symptom check and overload guard (and the finder itself if nothing else remains), so the oxlint rule resolves the receiver with `parserServices.getTypeAtLocation()` directly, as before PR #587. Keep `hasGrantsMethod` (nullish strip → Construct check → `grants` → `*Grants` suffix → method), which is not a workaround.
3. Flip the oxlint test cases that pin the skipped receivers from `valid` to `invalid`, so both suites assert the same outcome as the eslint suite: `topics[0]`, `registry["x"]` (string index signature yielding a Construct), `registry[0]`, `fn["k"]()` (reported only if the element type has `grants`), the literal-discriminated overload returning a Construct, and any further forms pinned as valid with a NOTE referring to the mis-resolution.
4. Remove the NOTEs that describe the guards as temporary mitigations, and update the `Known divergence` items in the rule documentation if any were added.
5. Verify: `vp check`, `vp test --run`, `vp run -r pack && bash scripts/integration-test.sh` (43 errors), CI green.

### Additional Context

```yaml
# machine-readable summary
rules: ["prefer-grants-property"]
affected_packages: ["oxlint-plugin-awscdk"]
kind: cleanup
introduced_by: ["#587"]
blocked_by: ["ubugeeei-prod/corsa-bind#490"]
trigger: "corsa-oxlint release that fixes corsa-bind#490 is pinned in packages/oxlint/package.json"
related_issues: ["#568"]
verified_on: "corsa-oxlint 1.14.0 (pinned by #597 since 2026-09-10) still mis-resolves index-signature access, non-literal index access, non-null assertions, literal-discriminated overloads and generic calls; literal array index (topics[0]) resolves correctly and its oxlint case already moved to invalid in #597"
```
